opnsense-freebsd/azuredeploy.json

```diff
@@ -0,0 +1,368 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "vmSize": {
+      "type": "string",
+      "defaultValue": "Standard_A1_v2",
+      "allowedValues":[
+        "Standard_A1_v2",
+        "Standard_A2_v2",
+        "Standard_A4_v2",
+        "Standard_B1s",
+        "Standard_B1ms",
+        "Standard_B2s",
+        "Standard_B2ms",
+        "Standard_B4ms",
+        "Standard_F1s",
+        "Standard_F2s",
+        "Standard_F4s"
+      ],
+      "metadata": {
+        "description": "VM size, please choose a size which allow 2 NICs."
+      }
+    },
+    "vmResourceName": {
+      "type": "string",
+      "defaultValue": "opnsense-vm",
+      "metadata": {
+        "description": "Resource name of VM. This will not be the VM hostname."
+      }
+    },
+    "vmStorageAccountType": {
+      "type": "string",
+      "defaultValue": "Standard_LRS",
+      "allowedValues": [
+        "Standard_LRS",
+        "Standard_GRS",
+        "Standard_ZRS",
+        "Premium_LRS"
+      ],
+      "metadata": {
+        "description": "Type of VM storage account."
+      }
+    },
+    "networkPrefix": {
+      "type": "string",
+      "defaultValue": "opnsense",
+      "metadata": {
+        "description": "Network resource prefix. Network resources will be named with this, i.e opnsense-routetable."
+      }
+    },
+    "vnetAddressPrefix": {
+      "type": "string",
+      "defaultValue": "10.0.0.0/23",
+      "metadata": {
+        "description": "Address prefix for VNET."
+      }
+    },
+    "publicSubnetAddressPrefix": {
+      "type": "string",
+      "defaultValue": "10.0.0.0/24",
+      "metadata": {
+        "description": "Address prefix for public subnet."
+      }
+    },
+    "privateSubnetAddressPrefix": {
+      "type": "string",
+      "defaultValue": "10.0.1.0/24",
+      "metadata": {
+        "description": "Address prefix for private subnet."
+      }
+    },
+    "publicNicIP": {
+      "type": "string",
+      "defaultValue": "10.0.0.5",
+      "metadata": {
+        "description": "Frontend private IP address which resides in public subnet."
+      }
+    },
+    "privateNicIP": {
+      "type": "string",
+      "defaultValue": "10.0.1.5",
+      "metadata": {
+        "description": "Backend private IP address which resides in private subnet."
+      }
+    },
+    "publicIPAddressType": {
+      "type": "string",
+      "defaultValue": "Dynamic",
+      "allowedValues": [
+        "Dynamic",
+        "Static"
+      ],
+      "metadata": {
+        "description": "Type of public IP address."
+      }
+    },
+    "domainNamePrefix": {
+      "type": "string",
+      "metadata": {
+        "description": "DNS Label for the Public IP. Must be lowercase. It should match with the following regular expression: ^[a-z][a-z0-9-]{1,61}[a-z0-9]$ or it will raise an error."
+      }
+    },
+    "_artifactsLocation": {
```

Comment:
need to add sasToken
See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#samples-that-contain-extra-artifacts-custom-scripts-nested-templates-etc) for how to handle artifacts